File: app/src/main/java/br/com/pedrosilva/tecnonutri/domain/interactors/GetFeedInteractor.java
package br.com.pedrosilva.tecnonutri.domain.interactors;


import java.util.List;

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem;
import br.com.pedrosilva.tecnonutri.domain.interactors.base.Interactor;


public interface GetFeedInteractor extends Interactor {

    interface Callback {
        void onFeedRetrieved(List<FeedItem> feedItems, int page, int timestamp);
        void onFeedRetrieveFailed(Throwable t);
    }
}

File: app/src/main/kotlin/br/com/pedrosilva/tecnonutri/domain/interactors/GetFeedInteractor.kt
package br.com.pedrosilva.tecnonutri.domain.interactors

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem
import br.com.pedrosilva.tecnonutri.domain.interactors.base.Interactor

interface GetFeedInteractor : Interactor {

    interface Callback {

        fun onFeedRetrieved(
            feedItems: List<FeedItem>,
            page: Int,
            timestamp: Int
        )

        fun onFeedRetrieveFailed(error: Throwable)
    }
}